pkg/agent/attestor/attestor_test.go

```diff
@@ -0,0 +1,169 @@
+package attestor
+
+import (
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus/hooks/test"
+
+	"crypto/x509"
+	"github.com/spiffe/spire/proto/agent/keymanager"
+	"github.com/spiffe/spire/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/proto/api/node"
+	"github.com/spiffe/spire/proto/common"
+	"github.com/spiffe/spire/test/mock/agent/catalog"
+	"github.com/spiffe/spire/test/mock/proto/agent/keymanager"
+	"github.com/spiffe/spire/test/mock/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/test/mock/proto/api/node"
+	"github.com/spiffe/spire/test/util"
+	"github.com/stretchr/testify/suite"
+	"path"
+	"testing"
+)
+
+type AttestorTestSuite struct {
+	suite.Suite
+
+	ctrl *gomock.Controller
+
+	attestor     Attestor
+	catalog      *mock_catalog.MockCatalog
+	nodeAttestor *mock_nodeattestor.MockNodeAttestor
+	keyManager   *mock_keymanager.MockKeyManager
+	nodeClient   *mock_node.MockNodeClient
+	config       *Config
+	expectation  *node.SvidUpdate
+}
+
+func (s *AttestorTestSuite) SetupTest() {
+	s.ctrl = gomock.NewController(s.T())
+
+	s.nodeAttestor = mock_nodeattestor.NewMockNodeAttestor(s.ctrl)
+	attestationData := &common.AttestedData{
+		Type: "join_token",
+		Data: []byte("foobar"),
+	}
+	fa := &nodeattestor.FetchAttestationDataResponse{
+		AttestedData: attestationData,
+		SpiffeId:     "spiffe://example.com/spire/agent/join_token/foobar",
+	}
+	s.nodeAttestor.EXPECT().FetchAttestationData(gomock.Any()).
+		Return(fa, nil)
+
+	svid, key, err := util.LoadSVIDFixture()
+	s.Require().NoError(err)
+
+	keyDer, err := x509.MarshalECPrivateKey(key)
+	s.Require().NoError(err)
+
+	s.keyManager = mock_keymanager.NewMockKeyManager(s.ctrl)
+	s.keyManager.EXPECT().FetchPrivateKey(gomock.Any()).Return(
+		&keymanager.FetchPrivateKeyResponse{PrivateKey: keyDer}, nil)
+	s.keyManager.EXPECT().GenerateKeyPair(gomock.Any()).Return(
+		&keymanager.GenerateKeyPairResponse{key.Y.Bytes(), keyDer}, nil)
+
+	s.catalog = mock_catalog.NewMockCatalog(s.ctrl)
+	s.catalog.EXPECT().NodeAttestors().
+		Return([]nodeattestor.NodeAttestor{s.nodeAttestor})
+	s.catalog.EXPECT().KeyManagers().
+		Return([]keymanager.KeyManager{s.keyManager})
+
+	s.nodeClient = mock_node.NewMockNodeClient(s.ctrl)
+	s.nodeClient.EXPECT().FetchBaseSVID(gomock.Any(), gomock.Any()).
```

Comment:
Feels a bit weird to have all these EXPECTs defined in SetupTest... Maybe it would be better to define them in the test cases which require them, or as a dedicated helper function that dependent test cases can call?